src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java

```diff
@@ -182,40 +181,94 @@ private SqlNode parseLiteralBool(final JsonObject expression) {
     }
 
     private List<SqlNode> parseExpressionList(final JsonArray array) {
-        assert array != null;
         final List<SqlNode> sqlNodes = new ArrayList<>();
         for (final JsonObject expr : array.getValuesAs(JsonObject.class)) {
-            final SqlNode node = parseExpression(expr);
-            sqlNodes.add(node);
+            sqlNodes.add(parseExpression(expr));
         }
         return sqlNodes;
     }
 
     private SqlGroupBy parseGroupBy(final JsonArray groupBy) {
-        if (groupBy == null) {
-            return null;
-        }
-        final List<SqlNode> groupByElements = parseExpressionList(groupBy);
-        return new SqlGroupBy(groupByElements);
+        return groupBy == null ? null : new SqlGroupBy(parseExpressionList(groupBy));
     }
 
     private SqlNode parseLiteralDate(final JsonObject exp) {
         final String date = exp.getString(VALUE);
         return new SqlLiteralDate(date);
     }
 
-    private SqlSelectList parseSelectList(final JsonArray selectList) {
-        if (selectList == null) {
-            return SqlSelectList.createSelectStarSelectList();
-        }
-        final List<SqlNode> selectListElements = parseExpressionList(selectList);
+    private SqlSelectList parseSelectList(final JsonArray selectList, final SqlNode from) {
+        final List<SqlNode> selectListElements = (selectList == null) ? collectAllInvolvedColumns(from)
+                : parseExpressionList(selectList);
         if (selectListElements.isEmpty()) {
             return SqlSelectList.createAnyValueSelectList();
         } else {
             return SqlSelectList.createRegularSelectList(selectListElements);
         }
     }
 
+    private List<SqlNode> collectAllInvolvedColumns(final SqlNode from) {
+        final List<SqlTable> involvedTables = collectInvolvedTables(from);
+        final Map<String, TableMetadata> tableMetadataMap = getInvolvedTablesMetadataMap();
+        final List<SqlNode> selectListElements = new ArrayList<>();
+        for (final SqlTable table : involvedTables) {
+            final String tableName = table.getName();
+            if (tableMetadataMap.containsKey(tableName)) {
+                final List<ColumnMetadata> columns = tableMetadataMap.get(tableName).getColumns();
+                for (int i = 0, columnsSize = columns.size(); i < columnsSize; ++i) {
+                    selectListElements.add(createColumn(i, table, columns.get(i)));
+                }
+            } else {
+                throw new IllegalStateException(
+                        "Unable to find a table metadata during collection involved columns for the table: "
```

Comment:
Use new error reporting.
Type: should not happen -> ticket

Reply:
We should do it separately https://github.com/exasol/virtual-schema-common-java/issues/195